api/src/main/java/graphql/nadel/Nadel.java

```diff
@@ -215,17 +218,17 @@ public GraphQLSchema getOverallSchema() {
 
         NadelExecutionParams nadelExecutionParams = new NadelExecutionParams(nadelExecutionInput.getArtificialFieldsUUID(), nadelExecutionInput.getNadelExecutionHints());
 
-        InstrumentationState instrumentationState = instrumentation.createState(new NadelInstrumentationCreateStateParameters(overallSchema, executionInput));
-        NadelInstrumentationQueryExecutionParameters instrumentationParameters = new NadelInstrumentationQueryExecutionParameters(executionInput, overallSchema, instrumentationState);
+        InstrumentationState instrumentationState = instrumentation.createState(new NadelInstrumentationCreateStateParameters(querySchema, executionInput));
+        NadelInstrumentationQueryExecutionParameters instrumentationParameters = new NadelInstrumentationQueryExecutionParameters(executionInput, querySchema, instrumentationState);
```

Comment:
Hmmm, wondering whether it's correct to pass in engine or query schema here. I'm leaning towards engine schema, but I see that this is also before any validation has occurred… hmmm.